File: kotlin-workshop/part-2-light-mocks/src/main/kotlin/gateway/ShippingGateway.kt
package gateway

import domain.ShippingConfirmation
import domain.ShippingRequest

/**
 * Gateway interface for shipping operations.
 * Injected into services - easy to mock!
 */
interface ShippingGateway {
    fun createShipment(request: ShippingRequest): ShippingConfirmation
    fun cancelShipment(trackingNumber: String): Boolean
}

File: typescript-workshop/part-4-behavior/src/gateway/ShippingGateway.ts
import { ShippingConfirmation } from '../domain/ShippingConfirmation';
import { ShippingRequest } from '../domain/ShippingRequest';

export interface ShippingGateway {
  createShipment(request: ShippingRequest): ShippingConfirmation;
  cancelShipment(trackingNumber: string): boolean;
}
